web3.js/src/transaction.ts

```diff
@@ -212,11 +248,45 @@ export class Transaction {
    */
   _json?: TransactionJSON;
 
+  // Construct a transaction with a blockhash and lastValidBlockHeight
+  constructor(opts?: TransactionBlockhashCtor);
+
+  // Construct a transaction with a nonce
+  constructor(opts?: TransactionNonceCtor);
+
+  /**
+   * @deprecated `TransactionCtorFields` has been deprecated and will be removed in a future version.
+   * Please use `TransactionBlockhashCtro` or `TransactionNonceCtor` instead.
+   */
+  constructor(opts?: TransactionCtorFields_DEPRECATED);
+
   /**
    * Construct an empty Transaction
    */
-  constructor(opts?: TransactionCtorFields) {
-    opts && Object.assign(this, opts);
+  constructor(
+    opts?:
+      | TransactionBlockhashCtor
+      | TransactionNonceCtor
+      | TransactionCtorFields_DEPRECATED,
+  ) {
+    if (!opts) {
+      return;
+      // eslint-disable-next-line no-prototype-builtins
+    } else if (opts.hasOwnProperty('recentBlockhash')) {
+      const oldOpts = opts as TransactionCtorFields_DEPRECATED;
+      Object.assign(this, oldOpts);
+      this.recentBlockhash = oldOpts.recentBlockhash;
+      // eslint-disable-next-line no-prototype-builtins
+    } else if (opts.hasOwnProperty('latestBlockhash')) {
```

Comment:
You meant this to be `lastValidBlockHeight`. It's probably worth writing some tests in `transaction.test.ts` to make sure that the right properties materialize on `Transaction` depending on the shape of the constructor args!

Reply:
Yep, I know what happened. When I originally wrote this I was using the lastValidBlockhash object rather than just pulling out the values. When I changed that I forgot to change the condition. Thanks for catching it!